tests/tests/postmessage-api.js

```diff
@@ -0,0 +1,148 @@
+// Sandstorm - Personal Cloud Sandbox
+// Copyright (c) 2015 Sandstorm Development Group, Inc. and contributors
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+"use strict";
+
+var utils = require('../utils');
+var actionSelector = utils.actionSelector;
+var appDetailsTitleSelector = utils.appDetailsTitleSelector;
+var very_short_wait = utils.very_short_wait;
+var short_wait = utils.short_wait;
+var medium_wait = utils.medium_wait;
+var very_long_wait = utils.very_long_wait;
+
+module.exports = {
+};
+
+module.exports['Install and launch test app'] = function (browser) {
+  browser
+    .loginDevAccount()
+    // test-3 introduces the JSON-formatted renderTemplate output
+    .url(browser.launch_url + '/install/072f8f84638d03e4de150e8fc4d3bd15?url=https://alpha-hlngxit86q1mrs2iplnx.sandstorm.io/test-3.spk')
+    .waitForElementVisible('#step-confirm', very_long_wait)
+    .click('#confirmInstall')
+    .url(browser.launch_url + "/apps/rwyva77wj1pnj01cjdj2kvap7c059n9ephyyg5k4s5enh5yw9rxh")
+    .waitForElementVisible(appDetailsTitleSelector, short_wait)
+    .assert.containsText(appDetailsTitleSelector, 'Test App')
+    .waitForElementVisible(actionSelector, short_wait)
+    .click(actionSelector)
+    .waitForElementVisible('#grainTitle', medium_wait)
+    .assert.containsText('#grainTitle', 'Untitled Test App test page')
+    .frame('grain-frame')
+      .waitForElementPresent('#randomId', medium_wait)
+      .assert.containsText('#randomId', 'initial state')
+    .frameParent()
+    .url(undefined, function (response) {
+      var grainUrl = response.value;
+      var expectedGrainPrefix = browser.launch_url + '/grain/';
+      browser.assert.ok(grainUrl.lastIndexOf(expectedGrainPrefix, 0) === 0, "url looks like a grain URL");
+      grainId = grainUrl.slice(expectedGrainPrefix.length);
+    });
+};
+
+var grainId = undefined;
+
+module.exports['Test setPath'] = function (browser) {
+  var expectedUrl = browser.launch_url + '/grain/' + grainId + '/#setpath';
+  browser
+    .frame('grain-frame')
+      .click('#setPath')
+      .pause(very_short_wait)
+    .frameParent()
+    .assert.urlEquals(expectedUrl);
+    // Should link-sharing links be expected to also hold the current path?
```

Comment:
I think probably the current path should be included when generating a link, but perhaps if the path is non-empty then we should have some fine-print that says: "Recipients will get access to the whole grain, not just /current/path."